internal/provider/types.go

```diff
@@ -0,0 +1,126 @@
+package provider
+
+import (
+	"fmt"
+	"math/big"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+type Image struct {
+	Digest   types.String `tfsdk:"digest"`
+	ImageRef types.String `tfsdk:"image_ref"`
+}
+
+type Manifest struct {
```

Comment:
You mentioned sharing with other providers...  Should these be under `internal`?

Reply:
Yeah I plan to move them out to `pkg` later, when the shape of them is a little more proven out. I'll probably test stuff locally with `apko_build` or `oci_append` first.